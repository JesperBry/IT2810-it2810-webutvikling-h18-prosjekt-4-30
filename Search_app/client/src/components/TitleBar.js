import React from "react";
import { Navbar, NavbarBrand, Nav, NavItem, NavLink } from "reactstrap";
import SkyLight from "react-skylight";

import SearchField from "./SearchField";

import "../styles/TitleBar.css";

class TitleBar extends React.Component {
  submit = values => {
    let obj = JSON.parse('{ "$search":' + JSON.stringify(values.search) + "}");
    console.log(obj);
  };

  render() {
    return (
      <div>
        <Navbar color="dark" dark>
          <NavbarBrand className="title" href="/">
            POLUTFORSKER
          </NavbarBrand>
          <Nav className="ml-auto" navbar>
            <NavItem>
              <NavLink onClick={() => this.simpleDialog.show()}>info</NavLink>
            </NavItem>
          </Nav>
        </Navbar>

        <SearchField onSubmit={this.submit} />

        <SkyLight
          dialogStyles={{ backgroundColor: "#ecf0f1" }}
          hideOnOverlayClicked
          ref={ref => (this.simpleDialog = ref)}
          title="Informasjon"
        >
          Her kan vi skrive informasjon om siden!
        </SkyLight>
      </div>
    );
  }
}

export default TitleBar;
