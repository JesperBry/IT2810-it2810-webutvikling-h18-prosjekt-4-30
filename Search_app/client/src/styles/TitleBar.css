@import url('https://fonts.googleapis.com/css?family=Roboto:300,400,500,700');

.ml-auto {
  margin-right: 2.5rem;
  cursor: pointer;
}

.navbar {
  height: 5rem;
}

.navbar-brand, .nav-link {
  margin-top: 0.5rem;
  position: absolute;
  top: 0;
}

.title {
  font-family: 'Roboto', sans-serif;
  font-weight: 300;
  margin-left: 0.5rem;
  letter-spacing: 0.2rem;
  font-size: 1rem;
}
