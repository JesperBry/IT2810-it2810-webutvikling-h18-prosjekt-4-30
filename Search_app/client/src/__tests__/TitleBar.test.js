import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { shallow } from "enzyme";

import TitleBar from "../components/TitleBar";

it("Matches snapshot", () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<TitleBar />);
  expect(result).toMatchSnapshot();
});

it("renders without crashing", () => {
  shallow(<TitleBar />);
});
