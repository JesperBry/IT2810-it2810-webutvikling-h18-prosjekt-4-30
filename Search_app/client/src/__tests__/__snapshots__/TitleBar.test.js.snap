// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div>
  <Navbar
    color="dark"
    dark={true}
    expand={false}
    tag="nav"
  >
    <NavbarBrand
      className="title"
      href="/"
      tag="a"
    >
      POLUTFORSKER
    </NavbarBrand>
    <Nav
      className="ml-auto"
      navbar={true}
      tag="ul"
      vertical={false}
    >
      <NavItem
        tag="li"
      >
        <NavLink
          onClick={[Function]}
          tag="a"
        >
          info
        </NavLink>
      </NavItem>
    </Nav>
  </Navbar>
  <ReduxForm />
  <SkyLight
    className=""
    closeButtonStyle={
      Object {
        "cursor": "pointer",
        "fontSize": "1.8em",
        "position": "absolute",
        "right": "10px",
        "top": "0px",
      }
    }
    closeOnEsc={true}
    dialogStyles={
      Object {
        "backgroundColor": "#ecf0f1",
      }
    }
    hideOnOverlayClicked={true}
    overlayStyles={
      Object {
        "backgroundColor": "rgba(0,0,0,0.3)",
        "display": "none",
        "height": "100%",
        "left": "0px",
        "position": "fixed",
        "top": "0px",
        "transitionProperty": "all",
        "transitionTimingFunction": "ease",
        "width": "100%",
        "zIndex": "99",
      }
    }
    showOverlay={true}
    title="Informasjon"
    transitionDuration={200}
  >
    Velkommen til Polutforsker! 
    <br />
    Nettsiden er basert på Vinmonopolets database, som finnes
    <a
      href="https://www.vinmonopolet.no/datadeling"
      target="_blank"
    >
       
      her.
    </a>
    <br />
     
    <br />
    På denne siden er det funksjonalitet for å søke etter varenavn og noen populære varetyper. 
    <br />
    <br />
    Søk på varenavn gjøres i søkefeltet, og returnerer alle produkter som inneholder minst ett av søkeordene. Varetype kan velges ved å trykke på «+»-symbolet.
    <br />
    <br />
    Videre kan søkeresultatet sorteres ved med valgene i dropdown-menyen.
    <br />
    <br />
    Utvidet produktinformasjon kan vises ved å trykke på produktnavnet. Her er det også mulighet for å legge til (uendelig mange) reviews av det valgte produktet.
  </SkyLight>
</div>
`;
